src/OpenLoco/src/GameCommands/VehiclePassSignal.cpp

```diff
@@ -0,0 +1,53 @@
+#include "Economy/Expenditures.h"
+#include "Entities/EntityManager.h"
+#include "GameCommands.h"
+#include "Types.hpp"
+#include "Vehicles/Vehicle.h"
+#include <OpenLoco/Interop/Interop.hpp>
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x004B0B50
+    static uint32_t vehiclePassSignal(EntityId headId, const uint8_t flags)
+    {
+        setExpenditureType(ExpenditureType::TrainRunningCosts);
+
+        try
```

Comment:
a try. Interesting whats the thinking behind it?

Reply:
Constructing `Train` can throw a std::exception